### PR TITLE
Fix various issues in fstab

### PR DESCRIPTION
**Pull request by gronke**
- prepend basejail nullfs mounts
- keep order of custom fstab entries
- bulk unmount nullfs mounts

### CODE REVIEW

File: iocage/lib/Jail.py [hunk 1]
```diff
@@ -991,29 +991,32 @@ def update_jail_state(self) -> None:
...
-        for mountpoint in mountpoints:
-            if os.path.isdir(str(mountpoint)):
-                iocage.lib.helpers.umount(
-                    mountpoint,
-                    force=True,
-                    logger=self.logger,
-                    ignore_error=True  # maybe it was not mounted
-                )
+        iocage.lib.helpers.umount(
+            ["-A", "-F", self.fstab.path, "-t", "nullfs"],
```
Comment (igalic) [inline iocage/lib/Jail.py]: can you please explain the options?

File: iocage/lib/LaunchableResource.py [hunk 2]
```diff
@@ -46,14 +46,23 @@ def create_resource(self) -> None:
...
+            )
+
     @property
-    def root_path(self):
+    def root_path(self) -> str:
         return self.root_dataset.mountpoint
 
     @property
     def root_dataset(self) -> libzfs.ZFSDataset:
         # ToDo: Memoize root_dataset
```
Comment (igalic) [inline iocage/lib/LaunchableResource.py]: TODO: open a bug for all TODOs we have

Comment (gronke) [reply iocage/lib/LaunchableResource.py]: > TODO: open a bug for all TODOs we have

for *each* ToDo 😎

File: iocage/lib/Jail.py (hunk 1 above)
Comment (gronke) [reply iocage/lib/Jail.py]: `-A` unmounts all entries from fstab
`-F` sets the provided path as fstab
`-t nullfs` ensures we only unmount nullfs mounts

Comment (igalic) [reply iocage/lib/Jail.py]: what about potential entries of other types?

Comment (gronke) [reply iocage/lib/Jail.py]: Finally you pointed me to an issue. It's important not to use `-A`, but `-a`. The latter removes the entries from fstab - `-A` removes all from the system :trollface:

**Pull request merged**